src/pubsub.ts

```diff
@@ -281,6 +282,33 @@ export class PubSub {
     }
   }
 
+  close(): Promise<void>;
+  close(callback: EmptyCallback): void;
+  /**
+   * Closes out this object, releasing any server connections. Note that once
+   * you close a PubSub object, it may not be used again. Any pending operations
+   * (e.g. queued publish messages) will fail. If you have topic or subscription
+   * objects that may have pending operations, you should call close() on those
+   * first if you want any pending messages to be delivered correctly. The
+   * PubSub class doesn't track those.
+   *
+   * @callback EmptyCallback
+   * @returns {Promise<void>}
+   */
+  close(callback?: EmptyCallback): Promise<void> | void {
+    const definedCallback = callback || (() => {});
+    if (this.isOpen) {
+      this.closeAllClients_()
+        .then(() => {
+          definedCallback(null);
+        })
+        .catch(definedCallback);
+      this.isOpen = false;
```

Comment:
I might move this to above `this.closeAllClients_`, just so it's clear that `this.isOpen` is already `false`, before the promise chain finishes executing.

Reply:
Good call, the whole thing should stop working if someone calls close() at all.